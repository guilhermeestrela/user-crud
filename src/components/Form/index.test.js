import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'

import { Form } from './'

describe('Form', () => {
    it('Should render Form correctly', () => {
        const { container, getByPlaceholderText } = render(<Form />)

        expect(container).toMatchSnapshot()
        expect(getByPlaceholderText(`Input glover's name`)).toBeInTheDocument()
    })

    it('Should be able to type a user name', () => {
        const { getByPlaceholderText } = render(<Form />)
        const input = getByPlaceholderText(`Input glover's name`);

        fireEvent.change(input, { target: {value: 'Test User'}})

        expect(input.value).toBe('Test User')
    })

    it('Should be able to add a user', () => {
        const mockedFunction = jest.fn()
        const { getByRole, getByPlaceholderText } = render(<Form addUser={mockedFunction}/>)
        const input = getByPlaceholderText(`Input glover's name`);
        const button = getByRole('button', { type: 'submit'})

        fireEvent.change(input, { target: {value: 'Test User'}})
        fireEvent.click(button)

        expect(mockedFunction).toBeCalledWith({
            name: 'Test User'
        })
    })
})
