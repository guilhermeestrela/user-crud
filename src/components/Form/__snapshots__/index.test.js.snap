// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Should render Form correctly 1`] = `
<div>
  <form>
    <input
      class="input"
      placeholder="Input glover's name"
      type="text"
      value=""
    />
    <button
      class="primary-button"
      type="submit"
    >
      ADD
    </button>
  </form>
</div>
`;
